courier/smtp.go

```diff
@@ -199,6 +200,18 @@ func (c *courier) dispatchEmail(ctx context.Context, msg Message) error {
 			// WithField("email_to", msg.Recipient).
 			WithField("message_from", from).
 			Error("Unable to send email using SMTP connection.")
+
+		if protoErr, success := err.(*textproto.Error); success && protoErr.Code >= 500 {
```

Comment:
It's probably better to use `errors.As` instead as that works with wrapper errors as well :)

Reply:
Right, done.